src/views/virtualmachines/tree/hooks/useTreeViewData.ts

```diff
@@ -35,17 +36,29 @@ export const useTreeViewData = (): UseTreeViewData => {
 
   const { featureEnabled: treeViewFoldersEnabled } = useFeatures(TREE_VIEW_FOLDERS);
   const [projectNames, projectNamesLoaded, projectNamesError] = useProjects();
+  const [namespaceNamesByCluster, namespaceNamesLoaded, namespaceNamesError] =
+    useMultiClusterNamespaces();
+  const { getResourceUrlOverride, supportsMulticluster } = useContext(KubevirtPluginContext);
 
   const [allVMs, allVMsLoaded] = useK8sWatchResource<V1VirtualMachine[]>({
     groupVersionKind: VirtualMachineModelGroupVersionKind,
     isList: true,
     limit: OBJECTS_FETCHING_LIMIT,
   });
 
+  const [multiclusterVms, multiclusterVmLoaded] = useKubevirtWatchResource<V1VirtualMachine[]>({
+    groupVersionKind: VirtualMachineModelGroupVersionKind,
+    isList: true,
+    limit: OBJECTS_FETCHING_LIMIT,
+    namespace: undefined,
+    namespaced: true,
+  });
+
+  // when using multicluster we don't have to worry about this because search api will return all namespaces
```

Comment:
I would suggest replacing the first block with the second, but keeping the name `allVMs` instead of `multiclusterVms`... basically just replace `useK8sWatchResource` with `useKubevirtWatchResource`.